pytorch_lightning/core/lightning.py

```diff
@@ -94,7 +95,7 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         torch._C._log_api_usage_once(f"lightning.module.{self.__class__.__name__}")
 
         # pointer to the trainer object
-        self.trainer = None
+        self.trainer: Optional[Trainer] = None
```

Comment:
```suggestion
        self.trainer: Optional["pl.Trainer"] = None
```
I'd rather do and remove the import